### PR TITLE
adds updating-resources

### PR DESCRIPTION
**Pull request by chrabyrd**
### brief description of changes

This adds a basic document to explain the process of automatically updating business data when a new version of a Graph is published. A link to this document will be referenced in [this branch](https://github.com/archesproject/arches/pull/9776) -- and the document will be updated with instructions on writing business data migrations once [this branch](https://github.com/archesproject/arches/pull/9701) is approved.

I'm unsure about proper format and if this document is living in the correct place, please feel free to update it or comment with changes!

This box **must** be checked
- [x] the PR branch was originally made from the base branch

This box **should** be checked
- [x] after these changes the docs build locally without error

This box **should only** be checked you intend to follow through on it (we can do it on our end too)
- [ ] I will `cherry-pick` all commits in this PR into other branches that should have them _after_ this PR is merged

### CODE REVIEW

File: docs/user-guide/updating-resources.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+##################
+Automatically Updating Resources
+##################
+
+Arches provides user interface features to update resource instances when the corresponding Graph changes. 
+
+
+Automatically update resource instances
+--------------------------------------------
+When publishing a new Graph, you are given the option to update the associated resource instances accordingly. This is a generic process that follows a specifc set of rules. It's good for handling minor changes, but for more complex changes you will likely want to write a business data migration.
```
Comment (ekansa) [inline docs/user-guide/updating-resources.rst:10]: I don't think we have any guidance  on writing migrations for Arches.  Given the abstractions and validation complexity of Arches, I'm sure this isn't straightforward! I'll make the "Writing custom Arches data migrations" a documentation TODO.

Anyway, in the mean time.  Let's add a note to the Django documentation on how to develop custom data migrations:
```

`(see Django documentation for how to write custom data migrations <https://docs.djangoproject.com/en/5.0/topics/migrations/#data-migrations>`_

```

File: docs/user-guide/updating-resources.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+The set of rules for this process is:
+
+- Changes made to a `Node`, `Card`, or `Widget` will change tile data as expected.
+- If you are updating the cardinality of a `Nodegroup` from `n` to `1`, the first `Tile` is preserved, the others are deleted.
+- The `ResourceInstance`'s `graph_publication_id` will be updated to the current `GraphXPublishedGraph` id. 
+
+
+The picture below illustrates the interaction for automatically updating resource instances:
+
+.. image:: ../images/update-resource-instances-modal.png
```
Comment (ekansa) [inline docs/user-guide/updating-resources.rst:21]: I'm a little unclear from this PR where the rules for the migration come from. The illustration appears to have a link to those rules. What information is at the other end of that link?

**Review by ekansa — CHANGES_REQUESTED** [review]
This looks like a very cool, important new set of functionality. I do have a few questions about some of the specifics about the migration rules however, and one suggestion about pointing to some additional Django documentation.

File: docs/user-guide/updating-resources.rst (hunk 2 above)
Comment (chrabyrd) [reply docs/user-guide/updating-resources.rst:21]: Oh yeah it's a chicken and egg thing. The link in this image should point to this document, but the link can't be created until the document is 😄

**Comment by ekansa** [comment]
Thanks for the updates @chrabyrd !

**Review by ekansa — APPROVED** [review]

**Pull request merged**